### ISSUE TITLE
Electron Wrapper

### ISSUE THREAD

**Issue by darkwater4213** [DestinyItemManager/DIM#9084]
### Proposed change

I'd like to build an Electron version of DIM, but... I'm not really sure where to start. In theory, I would like it to auto-update (showing off my ignorance of Electron here) whenever a new DIM release is pushed, as well as a "beta" option (self-explanatory). This isn't intended as a full-blown project and I'm not really interested in diving into the nitty-gritty of DIM's backend (sorry) to optimize and refine. My goal is really just to make PWAs (namely, DIM) work properly everywhere (not just Windows and mobile).

If you (the DIM team) were to take this and run with it (which I would _love_ to see), I would not mind overlays, especially in-game. Say you've input a query to DIM (ooh, a search overlay?). DIM could figure out where your stuff is in, say, your vault (use Bungie's API to get sort order and extrapolate from there ig) and give visual cues as to where the queried items are. Useful for dismantling times _en masse_.

But for now, **I just want to make a functioning-but-not-fancy DIM+Electron client, but I need someone to point me in the right direction**. :)

### How does this fit into your workflow?

Simply put, Windows is really the only one to properly handle Chrome apps (PWAs). Linux (Xfce4) tries to group your browser with them (and turning grouping off is... not good), MacOS doesn't display them at all--they just appear in a separate window without tabs. A proper app (albeit running on Electron) would also allow full Desktop integration without the user having to mess with poorly-documented CLI utils. This is exactly how Discord does it, and they've also got overlays and all sorts of fancy stuff. But you guys can handle that part ;)

**Comment by bhollis** [comment]
Hey there, thanks for the suggestion. Of course, this is something we've considered before, and it's not something the DIM team would work on, but feel free to give it a go. A few thoughts:

* DIM is an entirely client-side app - there's no backend involved (there's a DIM API for tags/loadouts storage but that's different). So there's no backend to mess with.
* Any Electron wrapper would need to wrap the live site, and not ship the code with the app. It's very important that we be able to update the site rapidly without having to make people download a new app. https://github.com/DestinyItemManager/iOS is a good example of this, and it's the only other "wrapper" we maintain (because you can't put PWAs into the App Store).
* The overlay idea is fun, but doesn't need to involve DIM at all. I suggest if you're interested in that, you just build a new app from scratch that uses the Bungie.net API, with the caveat that as far as we've found the API does not expose anything that would allow you to recreate the order of items as shown in game.
* PWAs (installed from Chrome) work great on macOS, showing their own icon in the Dock, their own entry in the tab switcher, and generally behaving like their own app. I don't know about Linux but there are enough window managers that I can't really comment.
* We don't really have any desktop integration we're interested in that PWAs don't allow through Chrome today.
* Dismantling items en masse is quite easy with DIM today - tag your items junk, use a search transfer to pull them all onto your character, and delete them all.

If you do manage to get this working as a shell around the live site, please let us know, we might be interested in taking it on. But I'm going to close this now as it isn't on the DIM team's roadmap.

**Comment by darkwater4213** [comment]
I decided that an electron wrapper would not be as helpful as a TUI version (lower resources and all that). I think I'll write it in Rust, using the DIM API. Would that work? Or should I use something like TS/JS with the Blessed library?
